wcfsetup/install/files/lib/data/moderation/queue/ModerationQueueReportAction.class.php

```diff
@@ -174,4 +174,31 @@ public function report()
             'reported' => 1,
         ];
     }
+
+    /**
+     * Validates the `changeJustifiedStatus` action.
+     *
+     * @since   5.4
+     */
+    public function validateChangeJustifiedStatus(): void
+    {
+        $this->queue = $this->getSingleObject();
+        if (!$this->queue->canEdit() || !$this->queue->canChangeJustifiedStatus()) {
+            throw new PermissionDeniedException();
+        }
+
+        $this->readBoolean('markAsJustified', true);
+    }
+
+    /**
+     * Removes this report by marking it as done without further processing.
+     */
+    public function changeJustifiedStatus(): void
+    {
+        $this->queue->update([
+            'additionalData' => \serialize([
+                'markAsJustified' => $this->parameters['markAsJustified'],
+            ]),
+        ]);
```

Comment:
Untested:
```suggestion
        $additionalData = $this->queue->additionalData;
        $additionalData['markAsJustified'] = $this->parameters['markAsJustified'];
        $this->queue->update([
            'additionalData' => \serialize($additionalData),
        ]);
```
Otherwise you might stomp `additionalData` that was added by some other user (e.g. a plugin).